dcm4che-core/src/main/java/org/dcm4che3/data/Attributes.java

```diff
@@ -2236,30 +2253,31 @@ public boolean addWithoutBulkData(Attributes other, BulkDataDescriptor descripto
             int vallen = (value instanceof byte[])
                     ? ((byte[])value).length
                     : -1;
-            if (descriptor.isBulkData(privateCreator, tag, vr, vallen, itemPointer))
+            if (descriptor.isBulkData(itemPointers, privateCreator, tag, vr, vallen))
                 continue;
 
             if (value instanceof Sequence) {
                 Sequence src = (Sequence) value;
-                setWithoutBulkData(privateCreator, tag, src, descriptor);
+                setWithoutBulkData(itemPointers, privateCreator, tag, src, descriptor);
             } else if (value instanceof Fragments) {
                 set(privateCreator, tag, (Fragments) value);
             } else {
-                set(privateCreator, tag, vr,
-                        toggleEndian(vr, value, toggleEndian));
+                set(privateCreator, tag, vr, toggleEndian(vr, value, toggleEndian));
             }
             numAdd++;
         }
         return numAdd != 0;
     }
 
-    private void setWithoutBulkData(String privateCreator, int tag, Sequence seq,
+    private void setWithoutBulkData(List<ItemPointer> itemPointers, String privateCreator, int tag, Sequence seq,
                                     BulkDataDescriptor descriptor) {
         Sequence newSequence = newSequence(privateCreator, tag, seq.size());
         for (Attributes item : seq) {
             Attributes newItem = new Attributes(bigEndian, item.size());
             newSequence.add(newItem);
-            newItem.addWithoutBulkData(item, descriptor);
+            itemPointers.add(new ItemPointer(privateCreator, tag, newSequence.size()));
```

Comment:
This add/remove allows the add without bulk data to run the the O(1) version of the lookup.